src/utils/patchMongoose.ts

```diff
@@ -0,0 +1,42 @@
+import mongoose, { IndexDefinition, IndexOptions } from 'mongoose';
+
+export function renameObjKey(
+  oldObj: Record<string, unknown>,
+  oldKey: string,
+  newKey: string | number
+) {
+  const keys = Object.keys(oldObj);
+  return keys.reduce(
+    (acc, val) => {
+      if (val === oldKey) {
+        acc[newKey] = oldObj[oldKey];
+      } else {
+        acc[val] = oldObj[val];
+      }
+      return acc;
+    },
+    {} as Record<string, unknown>
+  );
+}
+
+const oldMongooseSchemaIndexDef = mongoose.Schema.prototype.index;
```

Comment:
Is `mongoose` patching is crucial? Can we find another solution to avoid modifying alien module?

With such thing we can get a security violation.

Reply:
Yes I don't like this method - but was the only way to modify this method.

I have PR in mongoose here: https://github.com/Automattic/mongoose/pull/14042

Seems will be merged soon but still the issue will be in 7 unless this PR merged in both.

The patching is very small as well and if mongoose changed we can update to remove it.

Reply:
Why not create the correct index right away in the developer code?
Aliases are mongoose doc/code/operational level helpers. Indexes are DB/storage/data level staff. And mix aliases with indexes might bring a lot of headache on huge databases - it's quite easy rename alias in code, but it quite difficult to create/rename the real index in database. On my opinion, to be error prone protected it's better keep away aliases from indexes.

```diff
const UserSchema = new Schema(
  {
    n: {
      type: String,
      alias: 'name',
    },
    age: {
      type: Number,
    }
  }
)

- UserSchema.index({ name: 1, age: -1 });
+ UserSchema.index({ n: 1, age: -1 });
```

Reply:
I agree, this case only happen with aliases :) however if its working now I think it's good to keep it 😄